component_sdk/python/kfp_component/launcher/launcher.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fire
+import importlib
+import sys
+import logging
+
+def launch(file_or_module, args):
+    """Launches a python file or module as a command entrypoint.
+
+    Args:
+        file_or_module: it is either a file path to python file
+            a module path.
+        args: the args passed to the entrypoint function.
+
+    Returns:
+        The return value from the launched function.
+    """
+    try:
+        module = importlib.import_module(file_or_module)
+    except Exception:
+        try:
+            if sys.version_info.major > 2:
```

Comment:
Educate me if you will. Why calling spec_from_file_location and module_from_spec when import_module fails? In what situations will the following two function calls succeed in importing the module while the import_module fails.

Reply:
I see. Make sense.